src/foam/nanos/auth/CapabilityAuthService.js

```diff
@@ -248,17 +203,16 @@ foam.CLASS({
             }
 
             // Add the result to the cache
+            (( Map<String, Boolean> ) getCache()).put(agentKey, result);
             if ( result ) {
-              if ( agentKey != null ) (( Map<String, Boolean> ) getCache()).put(agentKey, result);
               return true;
             }
           }
-        } catch (Exception e) {
+        } catch ( Exception e ) {
           Logger logger = (Logger) x.get("logger");
           logger.error("check", permission, e);
         }
 
-        if ( result ) return true;
         maybeIntercept(x, permission);
```

Comment:
Do you think this is necessary? Since we do call this online 156?

Reply:
If we reached line 216 then we didn't call this method on line 156. Also, it means that there were no cashed false result before.